tests/services/test_barcode.py

```diff
@@ -82,3 +82,64 @@ def test_identify_tributos_code_line(self):
                     instance.barcode_image,
                     BarcodeService().generate_barcode_b64image(barcode),
                 )
+
+    def test_get_infos_from_comercial_barcode(self):
+        """
+        Dado:
+            - Uma lista de boletos comerciais
+        Quando:
+            - BarcodeService().get_infos_from_barcode_or_code_line(barcode)
+        Então:
+            - Deve ser retornado um dict com informações igual a expected
+        """
+        for barcode, code_line in self.cobrancas_barcodes_to_code_lines.items():
+            result = BarcodeService().get_infos_from_barcode_or_code_line(barcode)
+
+            expected = {
+                "instance": BarcodeService().identify(barcode),
+                "valid": True,
+                "barcode_number": barcode,
+                "code_line": code_line,
+                "type": "Comercial",
+                "bank": barcode[:3],
+                "amount": int(barcode[9:19]) / 100,
+                "due_date": BarcodeCobrancaService().calculate_due_date(barcode[5:9]),
+            }
+            self.assertEqual(result, expected)
+
+    def test_get_infos_from_tributo_code_line(self):
+        """
+        Dado:
+            - Uma lista de boletos tributários
+        Quando:
+            - BarcodeService().get_infos_from_barcode_or_code_line(code_line)
+        Então:
+            - Deve ser retornado um dict com informações igual a expected
+        """
+        for barcode, code_line in self.tributos_barcodes_to_code_lines.items():
+            result = BarcodeService().get_infos_from_barcode_or_code_line(code_line)
+
+            expected = {
+                "instance": BarcodeService().identify(barcode),
+                "valid": True,
+                "barcode_number": barcode,
+                "code_line": code_line,
+                "type": "Tributo",
+                "amount": int(barcode[4:15]) / 100,
+            }
+
+            self.assertEqual(result, expected)
+
+    def test_get_infos_from_a_wrong_barcode_or_code_line(self):
+        """
+        Dado:
+            - Um codigo errado
+        Quando:
+            - BarcodeService().get_infos_from_barcode_or_code_line(number)
+        Então:
+            - Deve ser retornado um ValueError com "Tipo não identificado!"
+        """
+        number = "12398523356"
+        with self.assertRaises(ValueError) as ctx:
+            BarcodeService().get_infos_from_barcode_or_code_line(number)
+        self.assertEqual(str(ctx.exception), "Tipo não identificado!")
```

Comment:
Seria interessante que a mensagem de erro fosse um pouco mais específica?
```suggestion
        self.assertEqual(str(ctx.exception), "Tipo não é um código de barra válido ou linha digitável válida! ")
```

Dependendo do contexto de onde essa mensagem aparecer (em outros projetos por exemplo) pode ser complicado entender/lembrar ao que essa mensagem se refere :thinking:

Reply:
Acho pertinente. Eu diria até que pode colocar O seguinte

```suggestion
        self.assertEqual(str(ctx.exception), "Código de barras ou linha digitável inválida!")
```